### DISCUSSION TITLE
Plotting with vertical exaggeration

### DISCUSSION THREAD

**Discussion by lperozzi** [pyvista/pyvista#2208] [Q&A]
## Description
I would like to plot a point dataset over a surface. I use this code:
```python
p = pv.Plotter(notebook=False)
p.add_mesh(grav, point_size=10.0, render_points_as_spheres=True, cmap='bwr',scalar_bar_args=sargs, clim=[-50,110],stitle='Residuals')
p.add_mesh(surf, cmap='gray')
p.show()
```
and I obtain this:

![image](https://user-images.githubusercontent.com/6198369/57818697-17650c80-7753-11e9-97ca-c1e104cd3b62.png)

Now I would like to add a vertical exageration factor of 3 to both surface and point set by using the `warp_by_scalar` method:

```python
grav = grav.warp_by_scalar(scale_factor=3.0)
surf = surf.warp_by_scalar(scale_factor=3.0)
```

the resulting plot is:

![image](https://user-images.githubusercontent.com/6198369/57818602-99086a80-7752-11e9-8292-a8e577f094b3.png)

It seems that the points set has not been scaled correctly. How to fix that?

## Example Data

[gravi.txt](https://github.com/pyvista/pyvista-support/files/3184973/gravi.txt)
[surf.txt](https://github.com/pyvista/pyvista-support/files/3184974/surf.txt)

**Comment by banesullivan** [comment]
Hi @lperozzi! This is an excellent question and outlines a common task when dealing with regional scale geo-datasets.

### Necessary Imports

```py
import numpy as np
import pyvista as pv
import PVGeo

pv.set_plot_theme('doc')
```

### Load the data

Looks like you have some data files in the UBC formats - PVGeo has readers for these!

```py
# load data files using PVGeo
gravi = PVGeo.ubc.GravObsReader().apply('gravi.txt')
psurf = PVGeo.ubc.TopoReader().apply('surf.txt')

# Filter points of topo to a surface
surf = psurf.delaunay_2d()
```

Reproduce the first figure:

```py
gargs = dict(point_size=5.0, render_points_as_spheres=True, 
           cmap='bwr', clim=[-50,110], stitle='Residuals')


p = pv.Plotter()
p.add_mesh(gravi, **gargs)
p.add_mesh(surf)
p.camera_position = [2,5,0.5]
p.show()
```

![download](https://user-images.githubusercontent.com/22067021/57825410-08368c80-775c-11e9-8a0b-49e93a0e94a8.png)


### The `warp_by_scalar` filter

This probably isn't the filter of choice for this particular example. The `warp_by_scalar` filter distorts the surface by the active scalar on the mesh by default - the current implementation above is warping the `grav` mesh of scattered points by the `'Residuals'` array which likely has a different dimension that the spatial reference in the z-axis.

### Directly editing the Z-coordinates


It may be best to directly edit the Z-coordinates of the meshes considering they share the same coordinate datum.

Note I use a factor of 10 instead of 3 to clearly demonstrate that the exaggeration is indeed occurring:
```py
scale_factor = 10.0
```

```py
# Apply a warp
warped_gravi = gravi.copy()
warped_gravi.points[:, -1] *= scale_factor
warped_surf = surf.copy()
warped_surf.points[:, -1] *= scale_factor

p = pv.Plotter(notebook=1)
p.add_mesh(warped_gravi,**gargs)
p.add_mesh(warped_surf)
p.camera_position = [2,1,0.2]
p.show()
```

![download](https://user-images.githubusercontent.com/22067021/57825691-336dab80-775d-11e9-8b4f-f4d382474e0e.png)



### Scaling the Plotting scene

This is a somewhat buggy feature we added that has some issues resulting from the VTK backend but it's possible to scale an entire rendering scene and not actually edit the spatial reference of the meshes. This is simply changing the visualization, not the meshes:

```py
p = pv.Plotter()
p.add_mesh(gravi, **gargs)
p.add_mesh(surf)
p.set_scale(zscale=scale_factor)
p.camera_position = [2,1,0.2]
p.show()
```

![download](https://user-images.githubusercontent.com/22067021/57825704-408a9a80-775d-11e9-8f56-50702e13090f.png)

**Comment by lperozzi** [comment]
Hi @banesullivan , many thanks for the hints. Actually scale an entire rendering scene is exactly what I need!

**Comment by banesullivan** [comment]
Just an FYI, there are issues when using the scaled render view: 

- https://discourse.vtk.org/t/scaling-a-rendering-scene/173
- https://github.com/pyvista/pyvista/issues/147
- https://github.com/pyvista/pyvista/issues/123
